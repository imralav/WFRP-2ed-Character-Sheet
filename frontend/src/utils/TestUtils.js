import React, {Fragment} from 'react';

import en from 'react-intl/locale-data/en';
import {addLocaleData, IntlProvider} from 'react-intl';
import renderer from 'react-test-renderer';

import {BrowserRouter as Router, Route} from 'react-router-dom';

addLocaleData(en);

class ComponentBuilder {
    constructor(component) {
        this.component = component;
    }

    withIntl() {
        this.component = (
            <IntlProvider locale="en" messages={{}}>
                <Fragment>
                {this.component}
                </Fragment>
            </IntlProvider>
        );
        return this;
    }

    withRouter() {
        this.component = (
            <Router>
                <Fragment>
                    {this.component}
                </Fragment>
            </Router>
        );
        return this;
    }

    build() {
        return renderer.create(this.component);
    }
}

export default class TestUtils {
    static buildComponent(component) {
        return new ComponentBuilder(component);
    }
}