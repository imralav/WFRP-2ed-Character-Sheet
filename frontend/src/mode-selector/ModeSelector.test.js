import React from 'react';
import ModeSelector from '../mode-selector';
import TestUtils from '../utils/TestUtils';

describe('ModeSelector', () => {
    let component;
    let componentRoot;
    let tree;
    beforeEach(() => {
        component = TestUtils.buildComponent(
            <ModeSelector/>
        ).withIntl().withRouter().build();
        componentRoot = component.root;
        tree = component.toJSON();
    });
    it('should match previous snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
});
