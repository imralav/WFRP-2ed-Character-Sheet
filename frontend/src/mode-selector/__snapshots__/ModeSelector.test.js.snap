// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeSelector should match previous snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-sm col-md-5"
  >
    <a
      className="modeSelectorTile text-center border border-dark rounded d-inline-flex flex-column"
      href="/character"
      onClick={[Function]}
    >
      <div>
        <img
          alt=""
          src="create-character.png"
        />
      </div>
      <div>
        <span>
          <span>
            Create character
          </span>
        </span>
      </div>
    </a>
  </div>
  <div
    className="col-12 col-sm offset-md-2 col-md-5 text-right"
  >
    <a
      className="modeSelectorTile text-center border border-dark rounded d-inline-flex flex-column"
      href="/character/open"
      onClick={[Function]}
    >
      <div>
        <img
          alt=""
          src="edit-character.png"
        />
      </div>
      <div>
        <span>
          <span>
            Use existing character
          </span>
        </span>
      </div>
    </a>
  </div>
</div>
`;
